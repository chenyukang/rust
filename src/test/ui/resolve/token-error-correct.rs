
// ignore-tidy-trailing-newlines
// Test that we do some basic error correction in the tokeniser.
// error-pattern: expected one of

fn main() {
    foo(bar(; //~ ERROR: expected expression
}
//~^ ERROR: mismatched closing delimiter: `}`

fn foo(_: usize) {} //~ ERROR: expected one of