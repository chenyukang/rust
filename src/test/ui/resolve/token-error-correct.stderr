error: mismatched closing delimiter: `}`
  --> $DIR/token-error-correct.rs:6:1
   |
LL |     foo(bar(;
   |            - unmatched opening delimiter begins here
LL |
LL | }
   | ^ mismatched closing delimiter

error: this file contains an unclosed delimiter
  --> $DIR/token-error-correct.rs:9:21
   |
LL | fn main() {
   |           - unclosed delimiter
LL |     foo(bar(;
   |        - unclosed delimiter
...
LL | fn foo(_: usize) {}
   |                     ^

error: this file contains an unclosed delimiter
  --> $DIR/token-error-correct.rs:9:21
   |
LL | fn main() {
   |           - unclosed delimiter
LL |     foo(bar(;
   |        - unclosed delimiter
...
LL | fn foo(_: usize) {}
   |                     ^

error: expected one of `)`, `,`, `.`, `?`, or an operator, found keyword `fn`
  --> $DIR/token-error-correct.rs:9:1
   |
LL | }
   |  - expected one of `)`, `,`, `.`, `?`, or an operator
...
LL | fn foo(_: usize) {}
   | ^^ unexpected token

error: aborting due to 4 previous errors

